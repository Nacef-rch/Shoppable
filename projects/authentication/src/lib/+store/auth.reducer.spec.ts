import * as authActions from '@authentication/+store/auth.actions';
import { initialState, authReducer as reducer } from '@authentication/+store/auth.reducer';
import * as userMock from '@authentication/mocks/user.mock.json';

describe('authReducer', () => {
    describe('undefined action', () => {
        it('should return the default state', () => {
            // GIVEN
            // WHEN
            const result = reducer(undefined, {} as any);
            // THEN
            expect(result).toMatchSnapshot();
        });

        describe('AUTHENTICATE', () => {
            it('`AUTHENTICATE_SUCCESS`, should add a user in the state', () => {
                // GIVEN
                const action = authActions.AUTHENTICATE_SUCCESS({
                    ...userMock.userGotToken,
                    expirationDate: new Date()
                });
                // WHEN
                const result = reducer(initialState, action);
                // THEN
                expect(result).toMatchSnapshot();
            });
        });
        it('`AUTHENTICATE_FAIL`, should add error in the state if there is error', () => {
            // GIVEN
            const errorMessage = 'Error';
            const action = authActions.AUTHENTICATE_FAIL({ errorMessage });
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('LOGIN/SIGNUP', () => {
        it('`LOGIN_START`, should set loading to true in state', () => {
            // GIVEN
            const action = authActions.LOGIN_START(userMock.userOnLogin);
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    it('`SIGNUP_START`, should set Loading to false in state', () => {
        // GIVEN
        const action = authActions.SIGNUP_START(userMock.userOnRegister);
        // WHEN
        const result = reducer(initialState, action);
        // THEN
        expect(result).toMatchSnapshot();
    });

    describe('LOGOUT', () => {
        it('should user state to initial state', () => {
            // GIVEN
            const action = authActions.LOGOUT();
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });

    describe('CLEAR_ERROR', () => {
        it('should reset state to initial state', () => {
            // GIVEN
            const action = authActions.CLEAR_ERROR();
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });

    describe('AUTO_LOGIN', () => {
        it('should reset state to initial state', () => {
            // GIVEN
            const action = authActions.AUTO_LOGIN();
            // WHEN
            const result = reducer(initialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
});
