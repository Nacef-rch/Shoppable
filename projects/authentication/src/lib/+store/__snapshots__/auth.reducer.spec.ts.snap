// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer \`SIGNUP_START\`, should set Loading to false in state 1`] = `
Object {
  "authError": null,
  "loading": true,
  "user": null,
}
`;

exports[`authReducer AUTO_LOGIN should reset state to initial state 1`] = `
Object {
  "authError": null,
  "loading": false,
  "user": null,
}
`;

exports[`authReducer CLEAR_ERROR should reset state to initial state 1`] = `
Object {
  "authError": null,
  "loading": false,
  "user": null,
}
`;

exports[`authReducer LOGIN/SIGNUP \`LOGIN_START\`, should set loading to true in state 1`] = `
Object {
  "authError": null,
  "loading": true,
  "user": null,
}
`;

exports[`authReducer LOGOUT should user state to initial state 1`] = `
Object {
  "authError": null,
  "loading": false,
  "user": null,
}
`;

exports[`authReducer undefined action \`AUTHENTICATE_FAIL\`, should add error in the state if there is error 1`] = `
Object {
  "authError": "Error",
  "loading": false,
  "user": null,
}
`;

exports[`authReducer undefined action AUTHENTICATE \`AUTHENTICATE_SUCCESS\`, should add a user in the state 1`] = `
Object {
  "authError": null,
  "loading": false,
  "user": User {
    "_token": "Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6Ijc0Mzg3ZGUyMDUâ€¦pefWhy_e1w88oc5VLXiXCdJV_ovfM_MgaooA4-YJZ-2PIybmQ",
    "_tokenExpirationDate": 2020-02-14T23:00:00.000Z,
    "email": "test@email.com",
    "id": "mQe3Nr6egJdYBHg1ETwEOtzq4FM2",
  },
}
`;

exports[`authReducer undefined action should return the default state 1`] = `
Object {
  "authError": null,
  "loading": false,
  "user": null,
}
`;
